xarray/backends/api.py

```diff
@@ -375,10 +375,11 @@ def open_dataset(
         scipy.io.netcdf (only netCDF3 supported). Byte-strings or file-like
         objects are opened by scipy.io.netcdf (netCDF3) or h5py (netCDF4/HDF).
     engine : {"netcdf4", "scipy", "pydap", "h5netcdf", "pynio", "cfgrib", \
-        "pseudonetcdf", "zarr"}, optional
+        "pseudonetcdf", "zarr"} or a custom backend class, optional
```

Comment:
How about `or a subclass of xarray.backends.BackenEntrypoint`?

That's copy/paste friendly and easy to google.

Reply:
we are constrained by the `numpydoc` type spec and `napoleon`'s parser so we would need to define a term for `subclass`, but otherwise that sounds good:
```suggestion
        "pseudonetcdf", "zarr"} or subclass of BackendEntrypoint, optional
```
Note that you can define a alias for `BackendEntrypoint` to translate to `~xarray.backends.BackendEntrypoint`

Reply:
This is going beyond my comfort zone. Is the current version ok? If not I'll take whatever you suggest.

Reply:
if you remove the `a` (and the quotes around `BackendEntrypoint`) this should be fine:
```suggestion
        "pseudonetcdf", "zarr"} or subclass of xarray.backends.BackendEntrypoint, optional
```

Reply:
Done.